packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -3544,3 +3579,417 @@ class _DropSliderValueIndicatorPathPainter {
     canvas.restore();
   }
 }
+
+/// The bar shape of a [Slider]'s thumb.
+///
+/// When the slider is enabled, the [ColorScheme.primary] color is used for the
+/// thumb. When the slider is disabled, the [ColorScheme.onSurface] color with an
+/// opacity of 0.38 is used for the thumb.
+///
+/// The thumb bar shape width is reduced when the thumb is pressed.
+///
+/// If [SliderThemeData.thumbSize] is null, then the thumb size is 4 pixels for the width
+/// and 44 pixels for the height.
+///
+/// This is the default thumb shape for [Slider]. If [ThemeData.useMaterial3] is false,
+/// then the default thumb shape is [RoundSliderThumbShape].
+///
+/// See also:
+///
+///  * [Slider], which includes an overlay defined by this shape.
+///  * [SliderTheme], which can be used to configure the overlay shape of all
+///    sliders in a widget subtree.
+class BarSliderThumbShape extends SliderComponentShape {
```

Comment:
This class name conflicts with one that exists in Google testing

Reply:
This was a we invented in the previous PR with Pierre. Naming stuff is hard :D 

Looking at this description. What If i call it `HandleThumbShape` or `VerticalBarThumbShape`?

<img width="972" alt="image" src="https://github.com/user-attachments/assets/c8c7c6c8-b7e5-4950-9c61-0e96f0012240">